plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -157,6 +158,9 @@ private static Object convertValue(Type type, Object value)
         else if (type instanceof VarbinaryType) {
             return Hex.encodeHexString(((Slice) value).getBytes());
         }
+        else if (type instanceof VarcharType) {
+            return ((Slice) value).toStringUtf8();
+        }
         return value;
```

Comment:
We are missing quite a lot of types here it seems?

Reply:
The rest of the supported types were already working, there are tests. This is just to add support for varbinary. For context the other supported types are primitives (arrays are not pushed down as pinot does not have a way to express an array literal): bigint, double, integer, real, boolean. Let me know if you have any suggestions.

Reply:
Continued at https://github.com/trinodb/trino/pull/9181#discussion_r705497387